### ISSUE TITLE
LightGCN double regularization on loss function

### ISSUE THREAD

**Issue by damicoedoardo** [RUCAIBox/RecBole#872]
Hello, 
while studying the repository I have noticed how for the particular model `LightGCN` there are two different L2 norm applied to the loss function which can lead to a wrong optimization process.

In particular we are adding: 
1) `weight_decay` when constructing the pytorch optimizer in the function _build_optimizer
2) `reg_weight` in `loss = mf_loss + self.reg_weight * reg_loss`

This lead to add 2 times the regularization in case both `reg_weight` and `weight_decay` are specified in the config file.

The solution that I would propose to solve that is to simply remove the param `reg_weight` and use `weight_decay` for the l2 regularization. 
I propose to act in this way so that the model is coherent with for example the BPR model which uses exactly `weight_decay` to define the amount of regularization.

Hope to be useful !

**Comment by 2017pxy** [comment]
@damicoedoardo Hi, thanks for your bug report and we have fixed this bug in #874.

**Comment by ChangxinTian** [comment]
Hi @damicoedoard , thanks for your feedback.
Setting `weight_decay` regularizes all trainable parameters, setting `reg_weight` regularizes some of the parameters flexibly.
We retain both parameters, but we do not recommend that you specify both at the same time. To do this, we have added a warning message in #874 .
